### PR TITLE
fix(pytest): revert "add an assert_error plugin method"

### PR DESCRIPTION
**Pull request by lengau**
Reverts canonical/craft-cli#218

Breaks #162 

Confirmed this is the cause. Compare:

- With #218: https://github.com/canonical/craft-application/actions/runs/12305359436/job/34344765176?pr=586
- Without: https://github.com/canonical/craft-application/actions/runs/12305776816/job/34346114053?pr=594

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -7,6 +7,11 @@ Changelog
 See the `Releases page`_ on GitHub for a complete list of commits that are
 included in each version.
 
+2.12.0 (2024-Dec-XX)
+--------------------
+
+- Remove the ``assert_error`` plugin method.
```
Comment (medubelko) [inline docs/changelog.rst:13]: Doing diligence: is there a replacement or fallback?

Comment (lengau) [reply docs/changelog.rst:13]: No, this is removing it because it was a breaking change.

Comment (lengau) [reply docs/changelog.rst:13]: Hmm though maybe:

```suggestion
- Remove the ``assert_error`` pytest plugin method. Use ``capsys`` to check for errors.
```

Comment (medubelko) [reply docs/changelog.rst:13]: The tense makes the grammar a bit ambiguous.

```suggestion
- Remove the ``assert_error`` pytest method. For checking errors, we recommend
  using ``capsys`` instead.
```

**Review by sergiusens — APPROVED** [review]
to me it is a 2.11.1, but I digress

**Comment by tigarmo** [comment]
> to me it is a 2.11.1, but I digress

if we want to get pedantic about it this would be a 3.0 actually

**Review by tigarmo — APPROVED** [review]

**Comment by lengau** [comment]
@tigarmo if we want to get really pedantic about it the only bug in 2.11 is that it wasn't called 3.0 ;-)

File: docs/changelog.rst (hunk 1 above)
Comment (lengau) [reply docs/changelog.rst]: Done!

**Review by medubelko — APPROVED** [review]

**Pull request merged**